### PR TITLE
analyze: verify verifies templates too

### PR DESCRIPTION
**Pull request by evverx**
Fixes:

```
$ systemd-analyze verify a@.service
Failed to load a@.service: Invalid argument
```

@keszybz , please, have a look.

### CODE REVIEW

File: src/analyze/analyze-verify.c [hunk 1]
```diff
@@ -30,6 +30,35 @@
...
+        if (!unit_name_is_valid(name, UNIT_NAME_ANY))
+                return -EINVAL;
+
+        if (unit_name_is_valid(name, UNIT_NAME_TEMPLATE)) {
+                r = unit_name_replace_instance(name, "i", &with_instance);
+                if (r < 0)
+                        return r;
+        }
+
+        if (with_instance)
```
Comment (evverx) [inline src/analyze/analyze-verify.c:64]: hm, I lose `path` here. will fix it.

**Comment by keszybz** [comment]
Oh, it seems that github doesn't loose force-push-replaced commits any more. That would be great news, we could stop opening new pull requests after comments!

I'm not sure about the "Failed to prepare filename %s:" message. Maybe it should be just "Invalid unit file name \"%s\":"? For example when I run this on some random file I get

```
Failed to prepare filename units/Makefile: Invalid argument
```

which just doesn't sound right. Anyway, this can be fixed in the future.

**Comment by evverx** [comment]
> I'm not sure about the "Failed to prepare filename %s:" message

Oh, I agree.

> it seems that github doesn't loose force-push-replaced commits any more.

Yeah, It works: ["Files Changed" -> comment](https://help.github.com/articles/commenting-on-the-diff-of-a-pull-request/)

But: [Why are my commits in the wrong order?](https://help.github.com/articles/why-are-my-commits-in-the-wrong-order/) :)

**Comment by zonque** [comment]
> Oh, it seems that github doesn't loose force-push-replaced commits any more. That would be great news, we could stop opening new pull requests after comments!

It never did if it was able to match previous hunks to the current version. Which is a valid attempt, but can lead to other confusion. If a comment is addressed differently in a force-pushed new version, the old comment still stays around.

The safest way to avoid all that is to file a new PR.

**Comment by evverx** [comment]
> The safest way to avoid all that is to file a new PR.

I agree.
But `push --force` helps sometimes: https://github.com/systemd/systemd/pull/2097#issuecomment-162072503

**Pull request merged**